diff --git a/tmk_core/protocol/usb_descriptor.c b/tmk_core/protocol/usb_descriptor.c
index 148daa27b..fd12de078 100644
--- a/tmk_core/protocol/usb_descriptor.c
+++ b/tmk_core/protocol/usb_descriptor.c
@@ -370,10 +370,10 @@ const USB_Descriptor_HIDReport_Datatype_t PROGMEM SharedReport[] = {
     HID_RI_USAGE(8, 0x80),                // System Control
     HID_RI_COLLECTION(8, 0x01),           // Application
         HID_RI_REPORT_ID(8, REPORT_ID_SYSTEM),
-        HID_RI_USAGE_MINIMUM(8, 0x01),    // Pointer
-        HID_RI_USAGE_MAXIMUM(16, 0x00B7), // System Display LCD Autoscale
-        HID_RI_LOGICAL_MINIMUM(8, 0x01),
-        HID_RI_LOGICAL_MAXIMUM(16, 0x00B7),
+        HID_RI_USAGE_MINIMUM(16, 0x0081), // System Power Down
+        HID_RI_USAGE_MAXIMUM(16, 0x008F), // System Restart
+        HID_RI_LOGICAL_MINIMUM(16, 0x0081),
+        HID_RI_LOGICAL_MAXIMUM(16, 0x008F),
         HID_RI_REPORT_COUNT(8, 1),
         HID_RI_REPORT_SIZE(8, 16),
         HID_RI_INPUT(8, HID_IOF_DATA | HID_IOF_ARRAY | HID_IOF_ABSOLUTE),
diff --git a/tmk_core/protocol/vusb/vusb.c b/tmk_core/protocol/vusb/vusb.c
index ea8ef35cc..333ac6723 100644
--- a/tmk_core/protocol/vusb/vusb.c
+++ b/tmk_core/protocol/vusb/vusb.c
@@ -590,10 +590,10 @@ const PROGMEM uchar mouse_extra_hid_report[] = {
     0x09, 0x80,             // Usage (System Control)
     0xA1, 0x01,             // Collection (Application)
     0x85, REPORT_ID_SYSTEM, //   Report ID
-    0x19, 0x01,             //   Usage Minimum (Pointer)
-    0x2A, 0xB7, 0x00,       //   Usage Maximum (System Display LCD Autoscale)
-    0x15, 0x01,             //   Logical Minimum
-    0x26, 0xB7, 0x00,       //   Logical Maximum
+    0x1A, 0x81, 0x00,       //   Usage Minimum (System Power Down)
+    0x2A, 0x8F, 0x00,       //   Usage Maximum (System Restart)
+    0x16, 0x81, 0x00,       //   Logical Minimum
+    0x26, 0x8F, 0x00,       //   Logical Maximum
     0x95, 0x01,             //   Report Count (1)
     0x75, 0x10,             //   Report Size (16)
     0x81, 0x00,             //   Input (Data, Array, Absolute)
