diff --git a/vllm/model_executor/layers/quantization/utils/marlin_utils.py b/vllm/model_executor/layers/quantization/utils/marlin_utils.py
index 87739efebb..539c64d01f 100644
--- a/vllm/model_executor/layers/quantization/utils/marlin_utils.py
+++ b/vllm/model_executor/layers/quantization/utils/marlin_utils.py
@@ -231,6 +231,13 @@ def check_moe_marlin_supports_layer(layer: LinearBase, group_size: int) -> bool:
     if current_platform.is_rocm():
         return False
+    capability_tuple = current_platform.get_device_capability()
+    device_capability = (
+        -1 if capability_tuple is None else capability_tuple.to_int()
+    )
+    if device_capability < 75:
+        return False
+
     hidden_size = layer.hidden_size
     intermediate_size_per_partition = layer.intermediate_size_per_partition
     # apply_router_weight_on_input is not supported for moe marlin
